### PR TITLE
[preferences] Filter events based on schema properties not actual values (fixes #1072)

### LINKED ISSUE #1072: Optional editor preferences are not taken into account
As the title said, if you try to provide a custom preference i.e "editor.lineNumbers" to "on" or "off", they're not taken into account because the implementation doesn't provide any default value for them. I noticed @svenefftinge changed some preferences to be optional a while back, which makes sense, but it also made it so optional preferences do not work anymore. I'm not exactly sure how to fix it yet, so I don't want to revert the change or anything.

### PR DESCRIPTION
**Pull request by svenefftinge**
Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

File: packages/preferences-api/src/common/preference-proxy.ts [hunk 1]
```diff
@@ -50,14 +52,18 @@ export function createPreferenceProxy<T extends Configuration>(preferences: Pref
...
-            if (p in configuration) {
-                const preference = preferences.get(p, configuration[p]);
-                if (validatePreference(schema, {
-                    [p]: preference
-                })) {
-                    return preference;
+            if (p in schema.properties) {
+                if (p in configuration) {
+                    const preference = preferences.get(p, configuration[p]);
+                    if (validatePreference(schema, {
```
Comment (akosyakov) [inline packages/preferences-api/src/common/preference-proxy.ts]: it looks bogus if `validatePreference` returns Promise

File: packages/preferences-api/src/common/preference-contribution.ts [hunk 2]
```diff
@@ -18,10 +18,21 @@ export const PreferenceSchema = Symbol("PreferenceSchema");
...
     properties: {
-        [name: string]: object
+        [name: string]: PreferenceProperty
     }
 }
 
+export interface PreferenceProperty {
+    description: string;
+    type?: JsonType | JsonType[];
+    minimum?: number;
```
Comment (epatpol) [inline packages/preferences-api/src/common/preference-contribution.ts]: Shouldn't we also have maximum as optional? I think I can see use cases where we would want a range of valid prefs.

Comment (svenefftinge) [reply packages/preferences-api/src/common/preference-contribution.ts]: There are many possible properties I haven't covered here. I will add a generic signature.

Comment (epatpol) [reply packages/preferences-api/src/common/preference-contribution.ts]: Yea I think it's fine

**Comment by svenefftinge** [comment]
The build failures seem unrelated.

**Pull request merged**